src/Resources/views/CRUD/base_array_macro.html.twig

```diff
@@ -8,17 +8,75 @@ For the full copyright and license information, please view the LICENSE
 file that was distributed with this source code.
 
 #}
-{% macro render_array(value, inline) %}
-    {% from _self import render_array %}
-    {% for key, val in value %}
-        {% if val is iterable %}
-            [{{ key }} => {{ render_array(val, inline) }}]
-        {% else %}
-            [{{ key }} => {{ val }}]
-        {% endif %}
-
-        {% if not loop.last and not inline %}
-            <br>
-        {% endif %}
-    {% endfor %}
+{# NEXT_MAJOR: move inline parameter under options #}
+{% macro render_array(value, inline, options = []) %}
```

Comment:
```suggestion
{% macro render_array(value, inline, options = {'key_translation_domain': false}) %}
```

Reply:
Why `not defined or false` condition can't solve issue?

Reply:
If I use `render_array(value, inline, {'value_translation_domain': true})`,  wouldn't I get an error since I lose the default value for `key_translation_domain` ?
Plus we need to be sure that  `render_array(value, inline, {'key_translation_domain': null})` is not overriden by the added default value.

The solution provided by @osavchenko avoid both of this.

Reply:
BTW, this solution will not work if at least one option will be passed here. It works only as a default value.

Reply:
I thought the issue was just with calling `render_array` with `{'key_translation_domain': null}`, isn't it the case? If we lose the default value, then it means we are not explicitely passing `null` for that key, and that's what @osavchenko said the issue was about. Regarding your other concerns, if macros behave the same as php functions do, then there will not be an override.

Reply:
If you want to add default value, since there is multiple you'll have to write
```
render_array(value, inline, options = {'foo': true, 'bar': false, 'baz': false})
```

Then, if I only override one option, the call will be 
```
render_array(value, inline, {'foo': null})
```
And I will lose the others default value, getting `error: bar is not defined`...

Reply:
:thinking: you're correct, we should use `{% set options = options|merge({"default": "value"}) %}` here I think

Reply:
That's a solution. But it should be `{% set options = defaults|merge(options) %}`

Reply:
Also, Twig has a null-coalesce operator… I don't want to think too much but I'm sure there must be some way to avoid these `if x is defined or …` everywhere.

Reply:
Thank you, guys! `merge` option works pretty good and I implemented it